### PR TITLE
Adds GraphQL

### PR DESCRIPTION
**Pull request by Naramsim**
Hi,

this PR brings in the ability to spin up a GraphQL server based on our PostgresQL tables. Currently the setup works only with the docker-compose installation. It's based on Hasura Engine (https://github.com/hasura/graphql-engine).

An hasura engine is started along pokeapi with docker compose. When pokeapi is built, Hasura can be started with `make` and all the metadata contained in the folder `/graphql` will be loaded in. In few seconds Hasura reads the pokeapi DB and exposes a custom private GraphiQL instance on port 8080. For more info browse this PR's [README](https://github.com/PokeAPI/pokeapi/blob/graphql/README.md).

4 GraphQL examples are also present in `/graphql/examples`.

---

Some README sections were moved.

### CODE REVIEW

File: apollo.config.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  client: {
+    service: {
+      name: "pokeapi",
+      url: "http://localhost:8080/v1/graphql",
+      headers: {
+        "x-hasura-admin-secret": "pokemon",
```
Comment (phalt) [inline apollo.config.js:7]: Secrets in code? I hope this isn't on our production version.

File: docker-compose.yml [hunk 2]
```diff
@@ -26,19 +28,40 @@ services:
...
+      - "8080:8080"
+    depends_on:
+      - "db"
+    restart: always
+    environment:
+      HASURA_GRAPHQL_DATABASE_URL: postgres://ash:pokemon@db:5432/pokeapi
+      HASURA_GRAPHQL_ENABLE_CONSOLE: "true"
+      HASURA_GRAPHQL_DEV_MODE: "false"
+      HASURA_GRAPHQL_ENABLED_LOG_TYPES: startup, http-log, webhook-log, websocket-log, query-log
+      HASURA_GRAPHQL_ADMIN_SECRET: pokemon
```
Comment (phalt) [inline docker-compose.yml:60]: secret - if this is just for dev that's fine

File: graphql/metadata/databases/default/tables/public_pokemon_v2_berryfirmness.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+array_relationships:
```
Comment (phalt) [inline graphql/metadata/databases/default/tables/public_pokemon_v2_berryfirmness.yaml:1]: aw man did you do all of these by hand? :heart:

File: docker-compose.yml (hunk 2 above)
Comment (Naramsim) [reply docker-compose.yml:60]: No, ofc :) This is just for dev. (They are hardcoded also for our Django and Postgres)

I didn't document our official productive setup but it's actually part of this repository. Basically, you need to start a different Compose file that reads a gitignored `.env` file. 

Here the docker compose: https://github.com/PokeAPI/pokeapi/blob/graphql/Resources/compose/docker-compose-prod-graphql.yml

Here the command to run: `docker-compose -f docker-compose.yml -f Resources/compose/docker-compose-graphql.yml up`

Our password in the productive system should be fine. it's in `~/pokeapi/.env`

File: graphql/metadata/databases/default/tables/public_pokemon_v2_berryfirmness.yaml (hunk 3 above)
Comment (Naramsim) [reply graphql/metadata/databases/default/tables/public_pokemon_v2_berryfirmness.yaml:1]: of course no, the UI comes with a dump functionality. I just cleaned and edited the files later. 😉 Hahha

Comment (phalt) [reply graphql/metadata/databases/default/tables/public_pokemon_v2_berryfirmness.yaml:1]: Hahaha nice.

**Review by phalt — COMMENTED** [review]
These changes look good to me. Do we promote it wider at the beta phase to test it out? Would be good to get some load testing done. I can use the twitter account to let people know.

**Comment by Naramsim** [comment]
Great Paul, I'll merge the PR. We could add something on Twitter and also add a banner on the official website.

**Comment by pokeapi-machine-user** [comment]
A [PokeAPI/api-data](https://github.com/PokeAPI/api-data) refresh has started. In 45 minutes the staging branch of [PokeAPI/api-data](https://github.com/PokeAPI/api-data/tree/staging) will be pushed with the new generated data. <br><br> The staging branch will be deployed in our [staging environment](https://staging.pokeapi.co/api/v2/) and you will be able to review the entire API. <br><br> A Pull Request ([master](https://github.com/PokeAPI/api-data/tree/master)<-[staging](https://github.com/PokeAPI/api-data/tree/staging)) will be also created at [PokeAPI/api-data](https://github.com/PokeAPI/api-data/pulls) and assigned to the PokeAPI Core team to be reviewed. If approved and merged new data will soon be available worldwide at [pokeapi.co](https://pokeapi.co/api/v2/).

**Comment by pokeapi-machine-user** [comment]
A [PokeAPI/api-data](https://github.com/PokeAPI/api-data) refresh has started. In 45 minutes the staging branch of [PokeAPI/api-data](https://github.com/PokeAPI/api-data/tree/staging) will be pushed with the new generated data. <br><br> The staging branch will be deployed in our [staging environment](https://staging.pokeapi.co/api/v2/) and you will be able to review the entire API. <br><br> A Pull Request ([master](https://github.com/PokeAPI/api-data/tree/master)<-[staging](https://github.com/PokeAPI/api-data/tree/staging)) will be also created at [PokeAPI/api-data](https://github.com/PokeAPI/api-data/pulls) and assigned to the PokeAPI Core team to be reviewed. If approved and merged new data will soon be available worldwide at [pokeapi.co](https://pokeapi.co/api/v2/).

**Pull request merged**